src/renderer.c

```diff
@@ -122,84 +167,352 @@ static int font_set_style(FT_Outline* outline, int x_translation, unsigned char
   return 0;
 }
 
-static void font_load_glyphset(RenFont* font, int idx) {
-  unsigned int render_option = font_set_render_options(font), load_option = font_set_load_options(font);
+static inline float font_get_whitespace_advance(RenFont* font, unsigned int codepoint) {
+  unsigned int em = font->face->size->metrics.x_ppem;
+  /**
+   * https://en.wikipedia.org/wiki/Whitespace_character
+   * Notes:
+   * 0. OGHAM SPACE MARK isn't implemented because it may contain lines
+   * 1. FIGURE SPACE, PUNCTUATION SPACE and IDEOGRAPHIC SPACE (CJK) isn't implemented correctly
+   *    because it requires actually loading something (this may cause infinite recursion)
+   * 2. MONGOLIAN VOWEL SEPARATOR is implemented as zero-width instead of narrow space
+   */
+  switch (codepoint) {
```

Comment:
Would it make sense to extract out of the switch the `0x0020` case (as I guess it's the most likely one?), and the ranges?

Reply:
I don't really understand what you meant by this.